experimental/pattern-matching-examples.md

```diff
@@ -0,0 +1,373 @@
+# AST examples of [Pattern Matching ESTree Proposal](proposal-pattern-matching-estree)
```

Comment:
Why not keep this as a section in the main `.md`?

Reply:
Oh the examples are not supposed to be part of the spec. They serves as an explainer only.